backend/src/main/kotlin/com/github/quillraven/toilapp/service/ToiletService.kt

```diff
@@ -32,9 +36,23 @@ class DefaultToiletService(private val toiletRepository: ToiletRepository) : Toi
             .flatMap { toiletRepository.save(toilet.copy(id = id)) }
     }
 
-    override fun getNearbyToilets(x: Double, y: Double, maxDistanceInMeters: Double): Flux<Toilet> {
+    override fun getNearbyToilets(x: Double, y: Double, maxDistanceInMeters: Double): Flux<ToiletResultDto> {
         LOG.debug("getNearbyToilets: (x=$x, y=$y, maxDistanceInMeters=$maxDistanceInMeters)")
-        return toiletRepository.getNearbyToilets(x, y, maxDistanceInMeters)
+        val position = Point(x, y)
+        val maxDistance = Distance(maxDistanceInMeters / 1000, Metrics.KILOMETERS)
+        //return toiletRepository.getNearbyToilets(x, y, maxDistanceInMeters)
+        val res = toiletRepository.findByLocationNear(position, maxDistance)
+        return res.map { r -> ToiletResultDto(r.content, distanceToMeter(r.distance))}
+    }
+
+    private fun distanceToMeter(distance: Distance): Double {
+        if(distance.metric == Metrics.KILOMETERS) {
+            return distance.value * 1000
+        } else if(distance.metric == Metrics.MILES) {
+            return distance.value * 1609.34
+        } else {
+            return distance.value * 1000
```

Comment:
You can use Kotlin's `when` for such cases which makes it easier to read in my opinion. Also, is it correct that `Neutral` Metric returns the same value as **KILOMETERS**. Shouldn't it be just `distance.value` ?

Example of when with lifted `return` keyword
```kotlin
private fun distanceToMeter(distance: Distance): Double {
        return when (distance.metric) {
            Metrics.KILOMETERS -> distance.value * 1000
            Metrics.MILES ->  distance.value * 1609.34
            else -> distance.value
        }
    }
```